### PR TITLE
Optimize passing tests computation

### PR DESCRIPTION
**Pull request by andre15silva**
- Test results are now stored as sets so operations on them, such as the passing tests computation, are closer to constant.
- Fix incoherence between JUnit5 and JUnit4 listeners, where JUnit4 wouldn't check `istTest()`.

Note that JUnit5 would allow for the test results to be stored immediately as passing, failing, etc, instead of having to do such computation, but the JUnit4 listener doesn't provide this information in `testFinished`, and `testFinished` is run after `testFailure` not allowing for such computation to be done while running the tests.  

~~To be rebased and merged after #113~~

### CODE REVIEW

**Review by danglotb — APPROVED** [review]
Could you give me a case where the `description.isTest()` would give `false`?

**Comment by andre15silva** [comment]
> Could you give me a case where the `description.isTest()` would give `false`?

Yes, as far as I understood it is `false` when the `description` represents a `TestSuite` (see http://junit.sourceforge.net/junit3.8.1/javadoc/junit/framework/TestSuite.html). From the javadoc of `Description`:

> A Description describes a test which is to be run or has been run. Descriptions can be atomic (a single test) or compound (containing children tests).

But now that I think of it, the listener already does that filtering for us, since any of those methods are only called for atomic tests. I'll revert those ifs.

File: src/main/java/eu/stamp_project/testrunner/listener/junit4/JUnit4TestResult.java [hunk 1]
```diff
@@ -49,38 +49,38 @@ public void testFinished(Description description) throws Exception {
...
                         failure.getException()
-                )
+			    )
         );
     }
 
     @Override
     public void testIgnored(Description description) throws Exception {
-        this.internalTestResult.getIgnoredTests().add(this.toString.apply(description));
+    	this.internalTestResult.getIgnoredTests().add(this.toString.apply(description));
```
Comment (danglotb) [inline src/main/java/eu/stamp_project/testrunner/listener/junit4/JUnit4TestResult.java:74]: Could you please indent properly this line?

Comment (andre15silva) [reply src/main/java/eu/stamp_project/testrunner/listener/junit4/JUnit4TestResult.java]: Done!

**Review by danglotb — APPROVED** [review]

**Pull request merged**